File: 03 C C++ Python R and Mojo review/03-3 R Language Review/01 Pointers/03.R
num <- 10
fnum <- 3.14

# Simulating the void pointer with a list containing references
vptr <- list(type = "int", value = num)
cat("Integer:", vptr$value, "\n")  # Output: 10

vptr <- list(type = "float", value = fnum)
cat("Float:", sprintf("%.2f", vptr$value), "\n")  # Output: 3.14

File: 03 C C++ Python R and Mojo review/03-2 Python Review/01 Pointers/03.py
# Python does not have pointers like C, but we can simulate similar behavior using references

num = 10
fnum = 3.14

# Simulating a void pointer behavior by storing references in a dictionary
vptr = {"type": "int", "value": num}
print(f"Integer: {vptr['value']}")  # Output: 10

vptr = {"type": "float", "value": fnum}
print(f"Float: {vptr['value']:.2f}")  # Output: 3.14